sections/sprint-4/updated-uppaal.tex

```diff
@@ -0,0 +1,144 @@
+The \uppTemp{TCP_Host} and \uppTemp{TCP_Client} will then begin to initialize a 3-way handshake to ensure that both parties are ready for transferring data between them.
+Once the handshake has been completed, the \uppTemp{TCP_Host} will send a message to the \uppTemp{Host} to inform that a new player has been connected via the \uppIn{TCP_client_connected} synchronization, and then enter an \uppLoc{Idle} location to indicate that it is ready to send messages.
+Likewise, in the \uppTemp{TCP_Client}, after the handshake is completed it will inform the \uppTemp{Client} that a connection has been successfully established, and enter an idle state where it is ready to receive messages.
+
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/TCP_client.png}
+    \caption{The TCP\_client template of the \uppaal model.}
+    \label{fig:sprint4-uppaal-tcp-client}
+\end{figure}
+
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/TCP_host.png}
+    \caption{The TCP\_host template of the \uppaal model.}
+    \label{fig:sprint4-uppaal-tcp-host}
+\end{figure}
+
+% UDP
+\subsection{UDP host templates}
+The final template seen in \autoref{fig:sprint4-uppaal-udp-host} is the \uppTemp{UDP_host}.
+It is very simple as it waits until the \uppOut{Host} communicates that the games has started through \uppIn{UDP_start}.
+When the \uppTemp{UDP_host} has started it waits for a \uppIn{UDP_send} synchronisation from the host and then notifies the \uppTemp{Clients} through the \uppOut{read_UDP_buffer} synchronisation that they should read the global buffer.
+\begin{figure}[h]
+    \centering
+    \includegraphics{sprint4/UDP_host.png}
+    \caption{The UPD\_host template of the \uppaal model.}
+    \label{fig:sprint4-uppaal-udp-host}
+\end{figure}
+
+% Trace example of the system
+\subsection{Happy simulation trace}
+A happy simulation trace can be seen on \autoref{fig:sprint4-happy-trace}.
+As seen on the trace \uppTemp{Client(0)} first connects through the threeway handshake without any timeouts.
+\uppTemp{Client(1)} Then also connects through threeway handshake without any timeouts.
+Finally the after both of the clients are in \uppLoc{Lobby} the \uppTemp{Host} transitions to \uppLoc{InGame}.
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=1\linewidth]{sprint4/connected-trace.png}
+    \caption{Example of a trace where both Clients connect without any problems}
+    \label{fig:sprint4-happy-trace}
+\end{figure}
+
+\subsection{Trace with a timeout}
+A trace where it times out once while connecting can be seen on \autoref{fig:sprint4-timeout-trace}.
+\uppTemp{Client(1)} connects without any issues, but \uppTemp{Client(0)} times out once while connecing.
+This can be seen on the trace of \uppTemp{TCP_client(0)}. When it sends the \uppOut{tcp_sync[0]} it timesout because it doesn't get are response from the \uppTemp{TCP_host(1)}.
```

Comment:
```suggestion
This can be seen on the trace of \uppTemp{TCP_client(0)}.
When it sends the \uppOut{tcp_sync[0]} it times out because it does not get any response from the \uppTemp{TCP_host(1)}.
```